modules/programs/kubecolor.nix

```diff
@@ -0,0 +1,82 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.kubecolor;
+  yamlFormat = pkgs.formats.yaml { };
+  inherit (pkgs.stdenv.hostPlatform) isDarwin;
+
+in {
+  meta.maintainers = with maintainers; [ ajgon ];
+
+  options.programs.kubecolor = {
+    enable = mkEnableOption "kubecolor - Colorize your kubectl output";
+
+    package = mkPackageOption pkgs "kubecolor" { };
+
+    enableAlias = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        When set to true, it will create an alias for kubectl pointing to
+        kubecolor, thus making kubecolor the default kubectl client.
+      '';
+    };
+
+    settings = mkOption {
+      type = yamlFormat.type;
+      default = { };
+      example = literalExpression ''
+        kubectl = lib.getExe pkgs.kubectl
+        preset = "dark";
+        paging = "auto";
+      '';
+      description = ''
+        Configuration written to {file}`$XDG_CONFIG_HOME/kube/color.yaml`
+        (Linux) or {file}`Library/Application Support/kube/color.yaml` (Darwin),
+        See <https://kubecolor.github.io/reference/config/> for supported
+        values.
+      '';
+    };
+  };
+
+  config = let
+    enableXdgConfig = !isDarwin || config.xdg.enable;
```

Comment:
Overriding the default configuration path with an XDG path should only happen when `config.home.preferXdgDirectories == true`.

When `config.home.preferXdgDirectories == false` I would expect the default configuration path to be used, i.e., the `KUBECOLOR_CONFIG` environment variable being unnecessary.